### ISSUE TITLE
Make Windows builds

### ISSUE THREAD

**Issue by nsg** [konstructs/client#33]
We need to make automatic Windows builds (triggered from Travis) to make sure that we maintain compatibility and the possibility to build Windows builds. We also need to find someone to test this and/or set up a test installation our self.

**Comment by Ajusa** [comment]
Hello!
I would glad to test this on my windows PC. I have recently been browsing Craft stuff and I like this fork and Daniel Oaks. If you want, I can test this.

**Comment by nsg** [comment]
Hi, sure! We have removed the curl dependency in this fork, and added zlib (to compress the chunk data) and openssl (for the login). I'm not a windows developer so I'm not completely sure what you need to do. If you are successful, report your findings here and will update the [README](https://github.com/nsg/Craft/blob/master/README.md), or even better, make a Pull Request :)

Maybe these?
http://gnuwin32.sourceforge.net/packages/zlib.htm
http://gnuwin32.sourceforge.net/packages/openssl.htm

If not, this is the official pages:
http://www.zlib.net/
https://www.openssl.org/

You need a server to run this fork, there is a server running at craft.nsg.cc that you can use, it's a dev server so the world will be deleted randomly :) If you prefer to test everything on localhost, see [nsg/Craft#server](https://github.com/nsg/Craft#server)

**Comment by Ajusa** [comment]
Okay, so managed to get it to compile. However, every time I try to connect to my localhost server, or to your server, I get gethostbyname: Result too large
I don't know what to do...

**Comment by nsg** [comment]
The problem is in [this function](https://github.com/nsg/Craft/blob/master/src/client.c#L271)
```
void client_connect(char *hostname, int port) {
if (!client_enabled) {
return;
}
struct hostent *host;
struct sockaddr_in address;
if ((host = gethostbyname(hostname)) == 0) {
perror("gethostbyname");
exit(1);
}
```

It is called from main.c at line [2794](https://github.com/nsg/Craft/blob/master/src/main.c#L2794):

```
 // CLIENT INITIALIZATION //
client_enable();
client_connect(g->server_addr, g->server_port);
client_start();
```

My guess is that the provided hostname is to large for some reason, at line [2771](https://github.com/nsg/Craft/blob/master/src/main.c#L2771) you finds this:

```
strncpy(g->server_addr, argv[1], MAX_ADDR_LENGTH);
```

Try to add a little debug after that line, something like:

```
printf("argv1: %s, server_addr: %s, max: %d\n",  argv[1], g->server_addr, MAX_ADDR_LENGTH);
```

**Comment by nsg** [comment]
@Ajusa I found and fixed the gethostbyname problem in #41 

I have also setup a cross compile of windows builds from travis, you will find the latest version build from master here: https://bintray.com/nsg/craft/client/view#files

**Comment by nsg** [comment]
Made a fix for Windows today, all compiled files are moved to https://bintray.com/konstructs/windows/client/view#files. We have also setup a webpage over here with download links: http://www.konstructs.org/download

I'm closing this issue.
